### ISSUE TITLE
readResolve function in serializable object false positive unused

### ISSUE THREAD

**Issue by ivan200** [detekt/detekt#5137]
for example i have object like this:
```kotlin
object Foo: Serializable
```
to be able to correctly serialize and deserialize it, i **must** write this:
```kotlin
object Foo: Serializable {
    private fun readResolve(): Any = Foo
}
```
As it mentioned here
https://blog.stylingandroid.com/kotlin-serializable-objects/
And for example here
https://stackoverflow.com/questions/56485742/not-possible-to-serialize-object-singleton-in-kotlin

## Expected Behavior
Detekt says there are no problems

## Observed Behavior
Detekt says "Private function readResolve is unused"

## Steps to Reproduce
Run detekt over the class above.

**Comment by marschwar** [comment]
Thank you for reporting this. You could add the `readResove` method to the list of allowed names.

```yaml
  UnusedPrivateMember:
    active: true
    allowedNames: '(_|ignored|expected|readResolve)'
```

**Comment by cortinico** [comment]
> Thank you for reporting this. You could add the `readResove` method to the list of allowed names.

Agree. I don't consider this a bug as this is framework specific + there is already a provided solution 👍 Closing
